crates/rust-analyzer/src/cli/analysis_stats.rs

```diff
@@ -310,7 +310,7 @@ pub fn analysis_stats(
 
 fn shuffle<T>(rng: &mut Rand32, slice: &mut [T]) {
     for i in (1..slice.len()).rev() {
-        let idx = rng.rand_range(0..i as u32) as usize;
+        let idx = rng.rand_range(0..(i + 1) as u32) as usize;
```

Comment:
```suggestion
        let idx = rng.rand_range(0..=i) as u32) as usize;
```

Reply:
```
   --> crates/rust-analyzer/src/cli/analysis_stats.rs:313:34
    |
313 |         let idx = rng.rand_range(0..=i as u32) as usize;
    |                                  ^^^^^^^^^^^^ expected struct `std::ops::Range`, found struct `std::ops::RangeInclusive`
    |
    = note: expected struct `std::ops::Range<u32>`
               found struct `std::ops::RangeInclusive<u32>`
```

Ranges are weird, see `RangeBounds`.

Reply:
Maybe stjepang could add this to `oorandom`, it's silly how it took us three tries to get it right :joy:.

Reply:
Yup, `fast_rand` gets this better, but I don't like the fact that it bundles thread-local rng, and that it's impossible to create rng not seeded with time

Reply:
Looks like it has a seed setter (weird), but yeah, the thread-local RNG can't be disabled.